### PR TITLE
New cannons

### LINKED ISSUE #75: Cannons
**_This issue is for both Art & Development Team_**

**Is your feature request related to a problem? Please describe.**
Cannons should become interactable, so implement the interactable abstract class for cannons. Additionally, use the new cannon model as per #65 

**Describe the solution you'd like**
Cannons are interactable, and will have enhanced features

@AoTTG-2/art-team 
- [ ] Cannon interactable/minimap sprites (Perhaps one sprite for both cannon types should suffice, or even an unique sprite for each of the two cannon types (ground cannon, wall cannon)
- [ ] Cannon sound effect

@AoTTG-2/development-team 
- [ ] Confirm that PR #74 is done
- [ ] Use the new cannon models from #65 
- [ ] Implement the "Interactable" abstract class for cannon (Both ground & wall cannons can use this), so that players can mount the cannon, and use it.
- [ ] Ground Cannon will have an extra interactable component (can be done by just adding the interactable component to only the ground cannon prefab), named "MoveCannonInteractable". This will allow the player to move this object. _We could also name it "MoveInteractable", which can be re-used to move other interactables in the future_

### PR DESCRIPTION
**Pull request by mwnDK1402**
- [x] Fix RC cannons
- [ ] Update their models
- [x] Make them use the Interaction Wheel
- [ ] Make them movable

Closes #75

### CODE REVIEW

File: Assets/Scripts/InstantiateTracker.cs [hunk 1]
```diff
@@ -85,19 +85,19 @@ public bool checkObj(string key, PhotonPlayer photonPlayer, int[] viewIDS)
...
+            case "rc resources/rc prefabs/cannonground":
                 if (PhotonNetwork.isMasterClient && !(FengGameManagerMKII.instance.allowedToCannon.ContainsKey(photonPlayer.ID) || FengGameManagerMKII.instance.restartingMC))
                 {
                     FengGameManagerMKII.instance.kickPlayerRC(photonPlayer, false, "spawning cannon item (" + key + ").");
                 }
                 return this.Instantiated(photonPlayer, GameResource.general);
 
-            case "rcasset/cannonwallprop":
-            case "rcasset/cannongroundprop":
+            case "rc resources/rc prefabs/cannonwallprop":
```
Comment (mwnDK1402) [inline Assets/Scripts/InstantiateTracker.cs:98]: This is an outdated path. Fix this if it is actually used.

File: Assets/Scripts/Room/LevelBuilder.cs [hunk 2]
```diff
@@ -17,6 +17,22 @@ public static List<Level> GetAllLevels()
 
     private static void AddClassicMaps()
     {
+        // Remove this before merging - Wagner
```
Comment (mwnDK1402) [inline Assets/Scripts/Room/LevelBuilder.cs:20]: Additional comment to remind myself.

Comment (mwnDK1402) [reply Assets/Scripts/Room/LevelBuilder.cs:20]: I think I'll just move this to the bottom.

**Comment by mwnDK1402** [comment]
Rewrite to-do list:
- [x] State Machine
- [x] Integrate with Interaction Wheel
- [x] Mount Player
- - [x] Control Cannon
- - [x] Control Player Position
- - [x] Unmount On Death
- [x] Mount Player Networking
- [x] Aim Cannon
- - [x] Rotation (requires new models)
- - [x] Rotation Constraints
- - [x] Curve Prediction
- [x] Aim Cannon Networking
- [x] Fire CannonBall
- - [x] CannonBall Factory
- - [x] Tunable Velocity
- - [x] Predictable Curve
- - [x] Particle Effects
- - [x] Kill Titans
- [x] Fire CannonBall Networking
- [ ] Move Cannon
- [ ] Move Cannon Networking

**Comment by Lithrun** [comment]
Closing this PR as it has been open since July without updates. Re-open once ready for Review.

**Pull request closed without merging**